Bugsnag/Assets/Bugsnag/Editor/SymbolUpload/BugsnagAndroidSymbolUploader.cs

```diff
@@ -0,0 +1,246 @@
+                    EditorUtility.ClearProgressBar();
+                    return; // Exit if the download or chmod process failed
+                }
+                cliExecutablePath = _cliDownloadPath;
+            }
+
+
+            var buildOutputPath = Path.GetDirectoryName(report.summary.outputPath);
+
+            var args = string.Format("upload unity-android --api-key={0} --verbose --project-root={1} {2}", apiKey, Application.dataPath, buildOutputPath);
+            if (!string.IsNullOrEmpty(config.AppVersion))
+            {
+                args += $" --version-name={config.AppVersion}";
+            }
+            if (config.VersionCode > -1)
+            {
+                args += $" --version-code={config.VersionCode}";
+            }
+            RunBugsnagCliCommand(cliExecutablePath, args);
+            EditorUtility.ClearProgressBar();
+        }
+
+        private string GetBugsnagCliDownloadUrl()
+        {
+
+            // Detect platform
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                return RuntimeInformation.OSArchitecture == Architecture.Arm64
+                    ? _cliDownloadUrl + "arm64-macos-bugsnag-cli"
+                    : _cliDownloadUrl + "x86_64-macos-bugsnag-cli";
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return RuntimeInformation.OSArchitecture == Architecture.X86
+                    ? _cliDownloadUrl + "i386-windows-bugsnag-cli.exe"
+                    : _cliDownloadUrl + "x86_64-windows-bugsnag-cli.exe";
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                return RuntimeInformation.OSArchitecture == Architecture.X86
+                    ? _cliDownloadUrl + "i386-linux-bugsnag-cli"
+                    : _cliDownloadUrl + "x86_64-linux-bugsnag-cli";
+            }
+
+            throw new PlatformNotSupportedException("Your platform is not supported for Bugsnag CLI.");
+        }
+
+        private bool DownloadBugsnagCli(string cliUrl, string downloadPath)
+        {
+            UnityEngine.Debug.Log("Starting Bugsnag CLI download process...");
+
+            var directory = Path.GetDirectoryName(downloadPath);
+            if (!Directory.Exists(directory))
+            {
+                Directory.CreateDirectory(directory);
+                UnityEngine.Debug.Log($"Created directory at {directory}");
+            }
+
+            if (File.Exists(downloadPath))
+            {
+                UnityEngine.Debug.Log($"Executable already exists at {downloadPath}, skipping download.");
+                return true;
+            }
+
+            UnityEngine.Debug.Log($"Downloading Bugsnag CLI from {cliUrl} to {downloadPath}...");
+
+            var downloadProcess = new Process()
+            {
+                StartInfo = new ProcessStartInfo
+                {
+                    FileName = "curl",
```

Comment:
I'm not sure we can assume the presence of curl? Do we need an extra assertion/error message that says to install curl or set the executable path for BugSnag.

Reply:
we decided that we can assume the presence of curl